### DISCUSSION TITLE
Migrating repos to this organization

### DISCUSSION THREAD

**Discussion by d-v-b** [zarr-developers/community#70] [General]
This discussion tracks proposals to migrate existing zarr-related projects to the `zarr-developers` github organization.

In a recent zarr community meeting, @TomNicholas said he would be interested in seeing his [VirtualiZarr](https://github.com/TomNicholas/VirtualiZarr) project belong to of the `zarr-developers` org.

Similarly, I would like [pydantic-zarr](https://github.com/janelia-cellmap/pydantic-zarr) to live in `zarr-developers` 

Do we have a formal procedure ingesting pre-existing repos?

**Comment by sanketverma1704** [comment]
Thanks for opening the discussion, @d-v-b.

After talking to @joshmoore, I learned that the repositories need to be transferred to someone on the [ZSC](https://github.com/zarr-developers/governance/blob/main/GOVERNANCE.md#zarr-steering-council). Then, the ZSC member would move the repository under `zarr-developers`.

> Do we have a formal procedure ingesting pre-existing repos?

AFAIK, there isn't a formal procedure for ingesting repositories and nothing of the sort is mentioned in the [governance](https://github.com/zarr-developers/governance/).

I think it'd be useful for us (especially in the long run) to mention the privileges, ownership and governance of the migrated repositories in the governance document somewhere.

**Comment by jhamman** [comment]
I offer two points for consideration here. 

1. The [Xarray-contrib model](https://github.com/xarray-contrib/xarray-contrib) could be a useful one to consider here. It lays out some basic attributes a project needs and a process by which projects can request to be moved into the organization.

2. We need to be very clear about what moving under the zarr-developers means in terms of governance (@MSanKeys963 also made this point). IMO, this is not well established for existing projects in the org. Moving a project like Virtualizarr into this org should mean that a) the project agrees to adopt the Zarr code of conduct and b) use a compatible license. The ZSC should be responsible for making sure those two criteria are followed. Beyond that, the developers of each project should maintain ownership (broadly defined, inc. admin permissions on the repo) over the project.

**Comment by joshmoore** [comment]
Sorry not sure how best to respond on GH discussions.

- I could definitely see starting with the contrib model especially if it lets us move fast right now. We could always move again afterwards.
- Even just for zarr-python, I think we need to update the governance model to make it clear that there are different decision-making groups (even if there's some overlap).
- That being said, if we agree the only requirement on acceptance of a repo are @jhamman's CoC & license, then that is manageable.
- Otherwise, it needs to be clear which decision-making group is taking responsibility for any repo that's being added (and retrospectively for a few).

All that being said, I'm general :100: on integrating the work from @d-v-b and @TomNicholas when/how we can! :tada:

**Comment by TomNicholas** [comment]
Hi guys, I would like to move forward here if we can.

> if we agree the only requirement on acceptance of a repo are @jhamman's CoC & license, then that is manageable.

> Moving a project like Virtualizarr into this org should mean that 
> a) the project agrees to adopt the Zarr code of conduct and 

I've read through the [Zarr org's code of conduct](https://github.com/zarr-developers/.github/blob/main/CODE_OF_CONDUCT.md), and am happy to abide by it. I can link to it from the virtualizarr docs too.

> b) use a compatible license. 

[VirtualiZarr currently has an Apache-2.0 license](https://github.com/TomNicholas/VirtualiZarr/blob/main/LICENSE), [zarr-python has an MIT licence](https://github.com/zarr-developers/zarr-python/blob/v3/LICENSE.txt), and the [zarr-developers org has a Creative Commons Zero Universal licence](https://github.com/zarr-developers/.github/blob/main/LICENSE). I don't know what's required of VirtualiZarr to be "compatible", but I would be happy for the project to adopt any of these.

> Otherwise, it needs to be clear which decision-making group is taking responsibility for any repo that's being added

I'm happy to collect a list of names of people who would constitute this group for VirtualiZarr if needed (or we can just use those who already have admin permissions on the repo).

Let me know what the next step is.

**Reply by joshmoore** [reply]
Starting this evening, I will be on the road for the next two weeks and don't want to block anyone. I'm personally happy moving forward with either/both of the two repos mentioned here with either (but only one) of the strategies: (1) accept the terms and move to the main org or (2) move to a contrib repo. Since I haven't done the latter, I'd say let's move forward with (1) via transferring the repos to one of the @zarr-developers/steering-council. In the background, we'll work on defining the procedure and if in doing so, we decide we *do* want a contrib org, then we can always migrate again.

**Reply by TomNicholas** [reply]
In that case @rabernat would you mind doing the honors of moving the repo? I think you should have the necessary permissions

**Reply by TomNicholas** [reply]
It's done! Thanks Ryan. @joshmoore let me know if there is anything else you think I should do (e.g. regarding licenses or similar)
